mlrun/serving/routers.py

```diff
@@ -701,3 +702,174 @@ def _init_endpoint_record(graph_server, voting_ensemble: VotingEnsemble):
             exc=exc,
             traceback=traceback.format_exc(),
         )
+
+
+class FeatureEnricher:
+    """enrich and impute real-time feature values from feature store
+
+    this is a temporary class for graph routers, code will move to feature store get_online_feature_service
+    """
+
+    def __init__(
+        self, feature_vector_uri, impute_policy: dict = {}, index_keys: list = None
+    ):
+        self.feature_vector_uri = feature_vector_uri
+        self.impute_policy = impute_policy
+        self.index_keys = index_keys
+
+        self._feature_service = None
+        self._impute_values = {}
+        self._feature_keys = None
+
+    def load(self):
+        """load the enricher: start the feature service and prep the imputing logic"""
+        self._feature_service = mlrun.feature_store.get_online_feature_service(
+            feature_vector=self.feature_vector_uri
+        )
+        vector = self._feature_service.vector
+        feature_stats = vector.get_stats_table()
+        self._impute_values = {}
+        self.index_keys = self.index_keys or vector.status.index_keys
+
+        self._feature_keys = list(vector.status.features.keys())
+        if vector.status.label_column in self._feature_keys:
+            self._feature_keys.remove(vector.status.label_column)
+
+        if "*" in self.impute_policy:
+            value = self.impute_policy["*"]
+            del self.impute_policy["*"]
+
+            for name in self._feature_keys:
+                if name not in self.impute_policy:
+                    if isinstance(value, str) and value.startswith("$"):
+                        self._impute_values[name] = feature_stats.loc[name, value[1:]]
+                    else:
+                        self._impute_values[name] = value
+
+        for name, value in self.impute_policy.items():
+            if name not in self._feature_keys:
+                raise ValueError(
+                    f"feature {name} in impute_policy but not in feature vector"
+                )
+            if isinstance(value, str) and value.startswith("$"):
+                self._impute_values[name] = feature_stats.loc[name, value[1:]]
+            else:
+                self._impute_values[name] = value
+
+    def enrich(self, inputs):
+        """enrich an input vector with features from the feature vector"""
+
+        # validate we have valid input struct
+        if (
+            not inputs
+            or not isinstance(inputs, list)
+            or not isinstance(inputs[0], (list, dict))
+        ):
+            raise ValueError("model inputs must be a list or list or list of dict")
```

Comment:
```suggestion
            raise ValueError("model inputs must be a list or list of dict")
```

Reply:
its list of list or list of dict (fixed)